airbyte-integrations/connectors/source-harvest/source_harvest/source.py

```diff
@@ -53,6 +54,11 @@ class SourceHarvest(AbstractSource):
     def get_authenticator(config):
         credentials = config.get("credentials", {})
         if credentials and "client_id" in credentials:
+            if "account_id" not in config:
+                raise AirbyteTracedException(
+                    "Config validation error: 'account_id' is a required property",
+                    failure_type=FailureType.config_error,
+                )
```

Comment:
General step-back question: Do we have a generalized exception we raise if the input is not valid against the JSON Schema?  (Could that handle this case if it did? I'm not actually sure, re: conditional requirements, but might work if it's a oneOf) 

Obviously we enforce this on the frontend in the platform but with terraform, pyairbyte etc. we should catch these things more generally if possible

Reply:
I think this make sense. I'm not sure if this was a possible in the whole workflow of the platform, but it is definitely possible from the source's perpective. Not sure if we should have something in the CDK to validate the config in the entrypoint before passing this to the source

Reply:
It looks like we do! We validate the config against the spec in `check` and then optionally at the beginning of discover and read https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/entrypoint.py#L171

Reply:
So if the spec is correct re: what is required and what isn't, a Config Error should be raised by [this validation](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/utils/schema_helpers.py#L175-L179)